ledger-core/v2/virtual/integration/conveyor/bargin_call_test.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/insolar/assured-ledger/ledger-core/v2/testutils/slotdebugger"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/injector"
+)
+
+type StepCallback func(ctx smachine.ExecutionContext)
+
+type SMWithCallback struct {
+	StepCallback StepCallback
+	CanStop      smsync.BoolConditionalLink
+}
+
+var dSMWithCallbackInstance smachine.StateMachineDeclaration = &dSMWithCallback{}
+
+type dSMWithCallback struct {
+	smachine.StateMachineDeclTemplate
+}
+
+func (*dSMWithCallback) InjectDependencies(_ smachine.StateMachine, _ smachine.SlotLink, _ *injector.DependencyInjector) {
+}
+
+func (*dSMWithCallback) GetInitStateFor(sm smachine.StateMachine) smachine.InitFunc {
+	s := sm.(*SMWithCallback)
+	return s.Init
+}
+
+func (s *SMWithCallback) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return dSMWithCallbackInstance
+}
+
+func (s *SMWithCallback) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	return ctx.Jump(s.stepProcess)
+}
+
+func (s *SMWithCallback) stepProcess(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	if s.StepCallback != nil {
+		s.StepCallback(ctx)
+	}
+	return ctx.Jump(s.stepDone)
+}
+
+func (s *SMWithCallback) stepDone(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	if ctx.Check(s.CanStop.SyncLink()) {
+		return ctx.Stop()
+	}
+	return ctx.Sleep().ThenRepeat()
+}
+
+func TestSMVDelegatedCallResponse_OK(t *testing.T) {
+	var (
+		mc  = minimock.NewController(t)
+		ctx = inslogger.TestContext(t)
+
+		requestRef    = gen.UniqueReference()
+		bargeInCalled = false
+		calleeCanStop = smsync.NewConditionalBool(false, "CanStop")
+	)
+
+	slotMachine := slotdebugger.New(ctx, t, false)
+	slotMachine.Start()
+	defer slotMachine.Stop()
+
+	{
+		calleeAction := func(ctx smachine.ExecutionContext) {
+			bargeIn := ctx.NewBargeInWithParam(func(param interface{}) smachine.BargeInCallbackFunc {
+				require.Nil(t, param)
+				bargeInCalled = true
+
+				return func(ctx smachine.BargeInContext) smachine.StateUpdate {
+					return ctx.WakeUp()
+				}
+			})
+			regResult := ctx.PublishGlobalAliasAndBargeIn(requestRef, bargeIn)
+			require.True(t, regResult)
+		}
+		smCallee := SMWithCallback{
+			StepCallback: calleeAction,
+			CanStop:      calleeCanStop,
+		}
+		smCalleeWrapper := slotMachine.AddStateMachine(ctx, &smCallee)
+		slotMachine.RunTil(smCalleeWrapper.BeforeStep(smCallee.stepDone))
+	}
+
+	{
+		callerAction := func(ctx smachine.ExecutionContext) {
+			slotLink, bargeInHolder := ctx.GetPublishedGlobalAliasAndBargeIn(requestRef)
+			require.False(t, slotLink.IsZero())
+			callResult := bargeInHolder.CallWithParam(nil)
+			require.True(t, callResult)
+		}
+		smCaller := SMWithCallback{
+			StepCallback: callerAction,
+			CanStop:      smsync.NewConditionalBool(true, "CanStop"),
+		}
+		smCallerWrapper := slotMachine.AddStateMachine(ctx, &smCaller)
+		slotMachine.RunTil(smCallerWrapper.AfterStop())
+	}
+
+	calleeCanStop.NewValue(true)
```

Comment:
How doest that work without ApplyAdjustment?

Reply:
But it work